src/index.js

```diff
@@ -311,63 +311,106 @@ class EventsCoreAPI {
+    const url = this.__getUrl(`/events/${consumerOrgId}/${projectId}/${workspaceId}/registrations/${registrationId}`)
     const sdkDetails = { requestOptions: requestOptions, url: url }
-    return this.__handleRequest(url, requestOptions, sdkDetails, codes.ERROR_CREATE_REGISTRATION)
+    return this.__handleRequest(sdkDetails, codes.ERROR_UPDATE_REGISTRATION)
   }
 
   /**
    * Get registration details for a given registration
    *
    * @param {string} consumerOrgId Consumer Org Id from the console
-   * @param {string} integrationId Integration Id from the console
+   * @param {string} projectId Project Id from the console
+   * @param {string} workspaceId Workspace Id from the console
    * @param {string} registrationId Registration id whose details are to be fetched
    * @returns {Promise<object>} Details of the webhook/journal registration
    */
-  getWebhookRegistration (consumerOrgId, integrationId, registrationId) {
+  getRegistration (consumerOrgId, projectId, workspaceId, registrationId) {
     const headers = {}
     const requestOptions = this.__createRequest('GET', headers)
-    const url = this.__getUrl(`/events/organizations/${consumerOrgId}/integrations/${integrationId}/registrations/${registrationId}`)
+    const url = this.__getUrl(`/events/${consumerOrgId}/${projectId}/${workspaceId}/registrations/${registrationId}`)
     const sdkDetails = { requestOptions: requestOptions, url: url }
-    return this.__handleRequest(url, requestOptions, sdkDetails, codes.ERROR_GET_REGISTRATION)
+    return this.__handleRequest(sdkDetails, codes.ERROR_GET_REGISTRATION)
   }
 
   /**
-   * Get all registration details for a given integration
+   * Get all registration details for a workspace
    *
    * @param {string} consumerOrgId Consumer Org Id from the console
-   * @param {string} integrationId Integration Id from the console
+   * @param {string} projectId Project Id from the console
+   * @param {string} workspaceId Workspace Id from the console
    * @returns {Promise<object>} List of all webhook/journal registrations
    */
-  getAllWebhookRegistrations (consumerOrgId, integrationId) {
+  getAllRegistrationsForWorkspace (consumerOrgId, projectId, workspaceId) {
     const headers = {}
     const requestOptions = this.__createRequest('GET', headers)
-    const url = this.__getUrl(`/events/organizations/${consumerOrgId}/integrations/${integrationId}/registrations`)
+    const url = this.__getUrl(`/events/${consumerOrgId}/${projectId}/${workspaceId}/registrations`)
     const sdkDetails = { requestOptions: requestOptions, url: url }
-    return this.__handleRequest(url, requestOptions, sdkDetails, codes.ERROR_GET_ALL_REGISTRATION)
+    return this.__handleRequest(sdkDetails, codes.ERROR_GET_ALL_REGISTRATION)
+  }
+
+  /**
+   * @typedef {object} Page
+   * @property {number} [page] page number to be fetched. Default 0 (optional)
+   * @property {number} [size] size of each page. Default 10 (optional)
+   */
```

Comment:
Why is this jsdoc snippet here?

Reply:
Only then it will be picked up when we generate the typings, so that it can prompt the user what is expected. Its the same for all such Object definitions ( HTTP Options, etc. )